sim-cli/src/parsing.rs

```diff
@@ -140,23 +188,94 @@ impl TryFrom<&Cli> for SimulationCfg {
     }
 }
 
+struct NodeMapping(HashMap<PublicKey, NodeInfo>, HashMap<String, NodeInfo>);
+
+pub async fn create_simulation_with_network(
+    cli: &Cli,
+    sim_params: &SimParams,
+    tasks: TaskTracker,
+) -> Result<Simulation, anyhow::Error> {
+    let cfg: SimulationCfg = SimulationCfg::try_from(cli)?;
+    let SimParams {
+        nodes: _,
+        sim_network,
+        activity,
+    } = sim_params;
+
+    // Convert nodes representation for parsing to SimulatedChannel
+    let channels = sim_network
+        .clone()
+        .into_iter()
+        .map(SimulatedChannel::from)
+        .collect::<Vec<SimulatedChannel>>();
+
+    let mut nodes_info = HashMap::new();
+    for channel in &channels {
+        let (pubkey, node_info) = channel.create_simulated_node();
+        nodes_info.insert(pubkey, node_info);
```

Comment:
I believe we have to return node info both `node_1` and `node_2` here?

If we have a simulated channel with:
```
   node_1: { pubkey: A }
   node_2: {pubkey: B }
```

And then we want to send payments:
```
 "activity": [
    {
      "source": "B",
      "destination": "A",
      "interval_secs": 1,
      "amount_msat": 2000
    }
]
```

As we have this now, we won't be able to look up the source node in our map?